contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,384 @@
+        IStrategy strategy = IStrategy(_strategyAddr);
+        address[] memory rewardTokenAddresses = strategy
+            .getRewardTokenAddresses();
+
+        if (rewardTokenAddresses.length > 0) {
+            strategy.collectRewardTokens();
+        }
+    }
+
+    /**
+     * @dev Swap all supported swap tokens for stablecoins via Uniswap.
+     */
+    function _swap() internal {
+        address[] memory allStrategies = IVault(vaultAddress)
+            .getAllStrategies();
+
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            _swap(swapTokens[i]);
+        }
+    }
+
+    /**
+     * @dev Swap a record token for stablecoins for Uniswap. The token must have
+     *       a registered price feed with the price provider.
+     * @param _swapToken Address of the token to swap.
+     */
+    function _swap(address _swapToken)
+        internal
+        returns (uint256[] memory swapResult)
+    {
+        RewardTokenConfig memory tokenConfig = rewardTokenConfigs[_swapToken];
+        require(
+            tokenConfig.uniswapV2CompatibleAddr != address(0),
+            "Swap token is missing token configuration."
+        );
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
+
+        IERC20 swapToken = IERC20(_swapToken);
+        uint256 balance = swapToken.balanceOf(address(this));
+        if (balance > 0) {
+            uint256 maxBalanceToSwap = balance;
+            if (tokenConfig.liquidationLimit != 0) {
+                maxBalanceToSwap = Math.min(
+                    balance,
+                    tokenConfig.liquidationLimit
+                );
+            }
+
+            // This'll revert if there is no price feed
+            uint256 oraclePrice = IOracle(priceProvider).price(_swapToken);
+            // Oracle price is 1e8, USDT output is 1e6
+            uint256 minExpected = (maxBalanceToSwap *
+                oraclePrice *
+                (1e4 - tokenConfig.allowedSlippageBps)).scaleBy( // max allowed slippage
+                6,
+                Helpers.getDecimals(_swapToken) + 8
+            ) / 1e4; // fix the max slippage decimal position
+
+            // Uniswap redemption path
+            address[] memory path = new address[](3);
+            path[0] = _swapToken;
+            path[1] = IUniswapV2Router(tokenConfig.uniswapV2CompatibleAddr)
+                .WETH();
+            path[2] = usdtAddress;
+
+            swapResult = IUniswapV2Router(tokenConfig.uniswapV2CompatibleAddr)
+                .swapExactTokensForTokens(
+                    maxBalanceToSwap,
+                    minExpected,
+                    path,
+                    address(this),
+                    block.timestamp
+                );
+
+            IERC20 usdt = IERC20(usdtAddress);
+            uint256 usdtBalance = usdt.balanceOf(address(this));
+            uint16 vaultBps = 1e4 - tokenConfig.harvestRewardBps;
+            require(
```

Comment:
We may want to have zero rewards. It's fine to remove this require.

Reply:
👍